packages/@aws-cdk/aws-sns-subscriptions/lib/sqs.ts

```diff
@@ -40,14 +40,20 @@ export class SqsSubscription implements sns.ITopicSubscription {
 
     // add a statement to the queue resource policy which allows this topic
     // to send messages to the queue.
-    this.queue.addToResourcePolicy(new iam.PolicyStatement({
+    const queuePolicyDependable = this.queue.addToResourcePolicy(new iam.PolicyStatement({
       resources: [this.queue.queueArn],
       actions: ['sqs:SendMessage'],
       principals: [snsServicePrincipal],
       conditions: {
         ArnEquals: { 'aws:SourceArn': topic.topicArn },
       },
-    }));
+    })).policyDependable;
+
+    // Add the subscription policy as a dependency for the subscription
+    const subscriptionDependencies: IDependable[] = [];
+    if (queuePolicyDependable) {
```

Comment:
You're setting this a few lines above. Under what conditions would this not be defined?

Reply:
It's not defined if the queue is imported or if `autoCreatePolicy` is set to false while no queue policy exists: https://github.com/aws/aws-cdk/blob/v1.159.0/packages/%40aws-cdk/aws-sqs/lib/queue-base.ts#L144